src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -283,6 +287,17 @@ We support
                         variants.Add(variant);
                     }
                 }
+
+                telemetryEnabled = configurationSection.GetValue<bool>("TelemetryEnabled");
+
+                IConfigurationSection telemetryMetadataSection = configurationSection.GetSection("TelemetryMetadata");
+
+                if (telemetryMetadataSection.Exists())
+                {
+                    telemetryMetadata = new Dictionary<string, string>();
+
+                    telemetryMetadataSection.Bind(telemetryMetadata);
```

Comment:
That `Bind` is going to require reflection. We have avoided it for all other settings. Seems worth to avoid it here as well.